solr/solrj/src/java/org/apache/solr/client/solrj/impl/LBSolrClient.java

```diff
@@ -699,11 +699,20 @@ public NamedList<Object> request(
         if (e.getRootCause() instanceof IOException) {
           ex = e;
           moveAliveToDead(wrapper);
-          if (justFailed == null) justFailed = new HashMap<>();
+          if (justFailed == null) {
+            justFailed = new HashMap<>();
+          }
```

Comment:
It's a shame to see duplication in getRootCause IOException detection with the dedicated clause.  Maybe it'd be cleaner to have another try-catch just around calling request() above that detects the IOException and throws it?  Just a thought; I leave it to you.